### PR TITLE
chore(QTDI-877): support conf/setting.xml in MavenDecrypter

### PR DESCRIPTION
**Pull request by ozhelezniak-talend**
https://qlik-dev.atlassian.net/browse/QTDI-877
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

**Review by undx — APPROVED** [review]
lgtm.
Thinking also that getter should not be an issue.
cliir maven plugin should be able to be configured to skip this in module.

**Review by undx — DISMISSED** [review]
See sonar comments. Mark'em as fixed for the relevants otherwise lgtm

File: component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java [hunk 1]
```diff
@@ -82,21 +100,63 @@ public Server find(final String serverId) {
...
-                .orElseGet(() -> new File(System.getProperty("user.home"), ".m2"));
+                .orElseGet(() -> new File(System.getProperty(USER_HOME), ".m2"));
+    }
+
+    private static List<File> findSettingsFiles() {
+        return Stream.of(
+                new File(getM2(), FILE_SETTINGS),
+                ofNullable(System.getenv(M2_HOME))
+                        .map(File::new)
+                        .map(it -> new File(it, "conf/" + FILE_SETTINGS))
```
Comment (undx) [inline component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java:156]: AFAIR, `conf/` is for maven home instead. Not sure it's in m2 home.

**Review by undx — DISMISSED** [review]
one small comment to check otherwise lgtm

File: component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java (hunk 1 above)
Comment (undx) [reply component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java:156]: yep, confirmed https://maven.apache.org/settings.html

Comment (ozhelezniak-talend) [reply component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java:156]: yes, and M2_HOME it's a path to maven installation directory
https://www.baeldung.com/java-maven-environment-variables
for maven 1 it was MAVEN_HOME
for maven 2 they renamed in to M2_HOME
what is for maven 3 - I don't know

Comment (ozhelezniak-talend) [reply component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java:156]: Okay, I see in that explanation, that they start to call it again MAVEN_HOME
I can change it to MAVEN_HOME
or we can support both
What do you think?

Comment (undx) [reply component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java:156]: As we handle multiple locations, let's put maven home as latest one.
Few chances that user uses that one...

**Review by undx — APPROVED** [review]

**Pull request merged**